### ISSUE TITLE
regression(test): diagnostics handler handler changedtick tracking should call handler only once if buffer changed in between callbacks

### ISSUE THREAD

**Issue by mochaaP** [nvimtools/none-ls.nvim#1]
```
Fail	||	diagnostics handler handler changedtick tracking should call handler only once if buffer changed in between callbacks	
            test/spec/diagnostics_spec.lua:126: Expected to be called 1 time(s), but was called 2 time(s)
            
            stack traceback:
            	test/spec/diagnostics_spec.lua:126: in function <test/spec/diagnostics_spec.lua:116>
```

Minimal repro:

```sh
make test-file FILE=test/spec/diagnostics_spec.lua
```

Seems a plenary / nvim update broke the async generator

**Comment by mochaaP** [comment]
addressed by commit 16bdde6a30b9611711e131e593a8fd418c429b09 for now

**Comment by mochaaP** [comment]
bisected to this commit: https://github.com/nvim-lua/plenary.nvim/commit/dd5a2f6d677f9c31dc55f0ab1ec2e9e43c408052
